pkg/i2gw/providers/common/utils.go

```diff
@@ -95,14 +95,18 @@ func NameFromHost(host string) string {
 	reg2, _ := regexp.Compile("^[^a-zA-Z0-9]+")
 	step2 := reg2.ReplaceAllString(step1, "")
 	// if nothing left, return "all-hosts"
-	if len(host) == 0 {
+	if len(host) == 0 || host == "*" {
 		return "all-hosts"
 	}
 	return step2
 }
 
 func RouteName(ingressName, host string) string {
-	return fmt.Sprintf("%s-%s", ingressName, NameFromHost(host))
+	routeName := ingressName
+	if host != "" {
+		routeName = fmt.Sprintf("%s-%s", ingressName, NameFromHost(host))
+	}
+	return routeName
```

Comment:
why?

Reply:
because if there is no host set the resulting name is `<ingress-name>-`

Reply:
oh ok, isnt it simpler to just wrap it in trimsuffix?

Reply:
the conditional needs to be there in any case, hence I do not see any difference.

Reply:
so I am probably missing something, why does it need to be there?

Reply:
oh I did not get your point initially. Yes, we can wrap it in a `trimsuffix`. Changed 👍

Reply:
so before this change, `routeName`used to be <ingressname>-all-hosts when`host==""`. and not it will be <ingressName> ? 
I dont mind just want to check I understand correctly and if we should not change other tests to align with this

Reply:
trimsuffix wont retain the same functionality, whats we want to have when host=="" ?